pkg/registry/resource/resourceclaim/strategy_test.go

```diff
@@ -587,6 +646,52 @@ func TestStrategyUpdate(t *testing.T) {
 				}
 			},
 		},
+		"keep-existing-fields-consumable-capacity": {
```

Comment:
I think we should have some additional test cases here:
* Make sure that adding these fields when the feature is enabled works.
* Make sure that existing fields remain even for updates with the feature off

Take a look at the cases covered for the PrioritizedList feature.